File: apps/api/internal/notes/note.go
// Package notes implements senior-scoped care notes (docs/03, CareNote).
package notes

import (
	"time"

	"github.com/google/uuid"
)

type Note struct {
	ID           uuid.UUID
	SeniorID     uuid.UUID
	AuthorUserID uuid.UUID
	AuthorName   string
	Content      string
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

File: packages/contracts/src/note.ts
/** A senior-scoped care note. */
export interface CareNote {
  id: string;
  seniorId: string;
  authorUserId: string;
  authorName: string;
  content: string;
  createdAt: string;
  updatedAt: string;
  canEdit: boolean;
}

export interface CareNoteListResponse {
  items: CareNote[];
}

export interface CreateCareNoteRequest {
  content: string;
}

export interface UpdateCareNoteRequest {
  content: string;
}
